detox/src/devices/common/drivers/android/tools/HashFileXfer.js

```diff
@@ -0,0 +1,20 @@
+const FileXfer = require('./FileXfer');
+const path = require('path');
+
+const HASH_PATH = '/data/local/tmp/detox';
+
+class HashFileXfer extends FileXfer {
```

Comment:
I don't understand this class, and why it inherits `FileXfer`:
1. `FileXfer`'s concern is very specific: Sending files into the device. Extending it in this way doesn't extend that concern, so it's shouldn't be done this way...
1. `FileXfer`'s methods `send` and `prepare` methods are never used in the context of hash storage. That is in fact also a testament to the first point.
1. Reading the hash file isn't about send a file, nor about pulling it - hence not about transferring file. The `readHashFile` method doesn't belong here.